[Portal Generated] Review request for Microsoft.Media to add version 2018-07-01

quintinb opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

quintinb committed:
Adds base for updating Microsoft.Media from version preview/2018-06-01-preview to version 2018-07-01

quintinb committed:
Adds base for updating Microsoft.Media from version preview/2018-06-01-preview to version 2018-07-01

quintinb committed:
Updates readme

quintinb committed:
Updates readme

quintinb committed:
Updates API version in new specs and examples

quintinb committed:
Updates API version in new specs and examples

quintinb committed:
Add initial updates for version 2018-07-01

quintinb committed:
Add initial updates for version 2018-07-01

quintinb committed:
Fix up issues found in initial run of the tooling

quintinb committed:
Fix up issues found in initial run of the tooling

quintinb committed:
Updates from linter warnings

quintinb committed:
Updates from linter warnings

quintinb committed:
Fixed issues with Media Services samples

quintinb committed:
Fixed issues with Media Services samples

quintinb committed:
Fixing up the md files

quintinb committed:
Fixing up the md files

quintinb committed:
Adding x-ms-pageable to the AccountFilters

quintinb committed:
Adding x-ms-pageable to the AccountFilters

quintinb committed:
Updated content-key-policies-list-all.json example

quintinb committed:
Updated content-key-policies-list-all.json example

quintinb committed:
Updated content-key-policies-list-by-lastModified.json to ensure required fields are in the response

quintinb committed:
Updated content-key-policies-list-by-lastModified.json to ensure required fields are in the response

quintinb committed:
Picked up some last minute updates to the 2018-07-01 version

quintinb committed:
Picked up some last minute updates to the 2018-07-01 version

quintinb committed:
Added back required properties in responses due to verification errors

quintinb committed:
Added back required properties in responses due to verification errors

quintinb committed:
Updates to the streaming service from comment review and adding the Live Event Input Access Control setting

quintinb committed:
Updates to the streaming service from comment review and adding the Live Event Input Access Control setting

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-node

AutorestCI commented:
# Automation for azure-sdk-for-go

AutorestCI commented:
# Automation for azure-sdk-for-java

viveklingaiah committed:
[GuestConfiguration] Include left out reportsV2 specs that is already approved (#3826)

hvermis committed:
[Datafactory] Adding new API to get GitHub access token (#3827)

* [ADF] Bringing back additionalProperties in Activity removed by mistake

* add github token api and exposure control api

* Removing Exposure Control API for now.

* Removing the example for ExposureControl API

* Fixed the example and made the response readonly

* Removing readonly from access token

mboersma committed:
[AKS] note that only one agent pool is currently supported (#3835)

mdo98 committed:
Azure Data Factory: Add swagger definition for specifying Spark environment variables in AzureDatabricksLinkedService (#3825)

* Azure Data Factory: Add swagger definition for specifying Spark environment variables in AzureDatabricksLinkedService.

* (nit: Capitalize sentence case in description of newClusterSparkConf and newClusterSparkEnvVars)

anuchandy commented on specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Accounts.json:
```diff
@@ -0,0 +1,910 @@
+      "required": [
+        "name"
+      ]
+    },
```
Looking at the api, the `Operation` model seems readonly model where its can be only read through `Operations.list`. If so should we mark all properties of this model as `readonly`? 

The new models representing the new properties of Operation are:

`MetricProperties`
`ServiceSpecification`
`Metric`

should we mark properties of all these models are `readonly`? given they are not used else where.

quintinb replied:
Yes, done.

anuchandy commented on specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Accounts.json:
```diff
@@ -0,0 +1,910 @@
+      "required": [
+        "nameAvailable"
+      ],
+      "description": "The response from the check name availability request."
```
`EntityNameAvailabilityCheckOutput` represents output of `Locations_CheckNameAvailability` operation, seems all properties can be readonly in this model. right?

anuchandy commented on specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AssetsAndAssetFilters.json:
```diff
@@ -0,0 +1,1264 @@
+        }
+      },
+      "type": "object",
+      "description": "The Streaming Locators associated with this asset."
```
are properties of `ListStreamingLocatorsResponse` and `AssetStreamingLocator` eligible to be marked as readonly? they are used to represents result of `Assets.listStreamingLocators` api and not used as payload for any create|update api.

quintinb replied:
Yes, I marked those as readonly.

anuchandy commented:
@quintinb 

1. Is this api-version deployed to production? if no, do you know when we are planning to?
2. please take a look at review comments.

@ravbhatnagar new api-version, please share ARM feedback.

zjpjack committed:
Fix ARMViolation Error for Microsoft.Authorization (#3844)

eladperets committed:
Add swagger spec for new resource type in Microsoft.PolicyInsights (#3815)

* Add swagger spec for new resource type in Microsoft.PolicyInsights

This commit adds a spec of a new preview resource type for the Microsoft.PolicyInsights- PolicyTrackedResources.
The new resource type is only avaliable in the 2018-07-01-preview API version.
The API itself is already deployed and passed an initial review by the ARM team.

* Add operations API to the spec and update readme.md

* Minor fixes

- Moved subscriptionId and apiVersion parameters to the end of the
sections.
- Removed the 'required' section from the definitions (irrelevant since
the API doesn't support PUT operations).
- Marked readonly properties.

q3blend committed:
Changing customConfig to a string (#3799)

* Updating customConfig to a string

* Updating customConfig to a string

* Updating swagger title for custom image search to Custom Image Search Client

* updating title from Custom Search API to client

refortie committed:
Change some string Microsoft.Logic modeled enums back into enums (#3775)

* Change some string modelled enums back into enums

* Change EDIFACT back to be an enum, not a string

viveklingaiah committed:
[GuestConfiguration] minor fixes to specs (#3851)

* [GuestConfiguration] minor fixes to specs

* changed configParameter list to array from object

grlin committed:
Grlin/updatingReqsforRA (#3852)

* adding principal type to putRA

* updating version to 2018-09-01-preview

* Updating requirements, request body IS required

* Updating 'required' fields in RA request body to match existing stable API requirements

* updating for 2018-01-01-preview and 2018-09-01-preview

balag0 committed:
Add new OsType parameters for Microsoft.Web/availablestacks (#3808)

jaredmoo committed:
Remove sensitivityLabels from sql readme.md (#3833)

There is a temporary issue with publishing the SDK containing this API, so it is removed from readme.md to avoid generating SDK.

ravbhatnagar commented:
@KrisBash from ARM to take a look. Kris - we can review this together.

MikhailTryakhov committed:
Local september branch merge to master (#3862)

* Trusted Root certificate (#3668)

* Rebased master to monthly network branch (#3687)

* Swagger Coverage/Completeness- for operation CheckNameAvailability (#3663)

* Update service.json

* CheckNameAvailability

* FIxed Parameter name

fixed couple of parameters name

* Update dns management Node.js version number to 3.0.0 (#3675)

* Update batchai Node.js package version to 2.0.2 (#3664)

* Update nofiticationhubs management Node.js package version to 2.0.0 (#3665)

* WebApps - Add BackupName to backup API models (#3560)

* WebApps - Add BackupName to replace non-ARM-compliant Name property on Backup API models

* Revert breaking change to BackupItem. BackupItemName property will be fine in place of BackupName for now.

* Update authorization management Node.js package version to 5.0.1 (#3676)

* Update storagesync management Node.js package version to 1.0.0 (#3677)

* add checknameavailability (#3608)

* add checknameavailability

* move checkname parameter to definition section

* fix brackets

* fix parameter name in checknameavailability exanple

* update code owners for PowerBIDedicated

* New Batch data plane API version 2018-08-01.7.0 (#3657)

* Add new batch version 2018-08-01.7.0

* Update the new API

* Fix up output from OAD, it does not output valid JSON

* update OAV (#3678)

* Update storage management Node.js package version to 5.2.0 (#3679)

* Update compute.json and disk.json for some issues and add zone and rolling upgrade. (#3642)

* Breaking change tool correctly handles multiple files with same name

* Fix to breaking changes table generation

* Trusted Root certificate (#3668)

* Updated Microsoft.Network version to 2018-08-01 (#3672)

* Adds base for updating Microsoft.Network from version stable/2018-07-01 to version 2018-08-01

* Updates readme

* Updates API version in new specs and examples

* Added AddressPrefixes on Subnet properties (#3714)

* Added address prefixes to subnet

* Added example of multiple address prefixes.

* Python Network conf 2018-08-01

* Changes to include required properties for outbound rule (#3728)

* Mark outbound rule properties as required

* Update outbound rule with required properties

* Removing old change

* Fix Python 2018-08 conf

* Use only PublicIpAddress in Azure Firewall, not both Public and InternalPublic (#3743)

* Do not confuse users with both InternalPublicIp and PublicIp for the same thing

* Update AzureFirewall PUT example

* This closes #3474

* Use the correct name for publicIPAddress in AzureFirewall PUT example

* Azure Firewall FQDN Tag top level resource (#3744)

* Azure Firewall FQDN Tag top level resource

* Name FqdnTags property in AzureFirewallApplicationRule properly

* Empty commit to trigger new validation

* Vtapfinal (#3684)

* Swagger Coverage/Completeness- for operation CheckNameAvailability (#3663)

* Update service.json

* CheckNameAvailability

* FIxed Parameter name

fixed couple of parameters name

* Adds base for updating Microsoft.Network from version stable/2018-07-01 to version 2018-08-01

* Updates readme

* Updates API version in new specs and examples

* vtap update

* VTAP changes

* VTAP changes

* Adding examples for VTAP resource

* Update changes for Examples

* Further fixes to examples

* Add exception for vtap's RequiredPropertiesMissingInResourceModel in readme

* Azure Firewall NAT Rule Collection (#3745)

* NAT Rule Collections for Azure Firewall

* Update examples to include the new NAT Rule Collection

* Shorten the NAT RC Properties name and mark etag read-only

* Add NAT Rule Collection to Azure Firewall properties

* Fix all <<a Azure>> in helper messages

* Do not allow ICMP as Network protocol (#3742)

* Adding linkedResourceType and hostedWorkloads to nic (#3780)

* Adding support for interface endpoints (#3783)

* Creating Interface Endpoint definition

* Adding references to interface endpoints in existing resources

* Added Trusted root certificate changes to new API version (#3731)

* Add ExpressRoute gateway resource (#3776)

* Initial version

* Refactor connections.

* Update API version

* Add enum values.

* Addressed all comments.

* Adding support for delegations on a subnet (#3805)

* Adding subnet delegation property on VNET

* Updating subnet/vnet create & get examples to reflect delegation

* Adding support for GetAvailableDelegations operation

* Fixing properties in Swagger & updating examples to reflect fixes

* Add availableDelegation as a subResource

* Remove type from examples

* Further updates to delegation PR

* Additional changes to structure

* Correct issues for SDK generation (#3818)

* Remove x-ms-client-flatten attribute as that results in wrong
descriptiosn during generation of the SDKs.

* Missing descriptions.

* Rename TargetUrls to TargetFqdns (#3739)

* Add Network Profile swagger and examples (#3806)

* add network profile swagger and examples

* add x-ms-client-flatten property as per feedback

* appease travis by fixing model + examples

* vtapswaggerupdate (#3837)

* Add Service Association Link and IP Configuration Profiles on Subnet (#3840)

* add service association links for virtual network

* add ipconfig profiles on subnet

* fixed example issue

* Review for tabfix and properties fromat rename

* Vtap swagger update and made the property as readonly (#3845)

* vtapswaggerupdate

* Adding readonly property

* Updated AppGW Autoscale Configuration (#3838)

* VPN gateway new API for Reset VPN client PSK (#3800)

* Brooklyn Ignite Swagger changes (#3715)

* Brooklyn Ignite Swagger changes

* Add missing example file changes

* Add virtual hub route table

* Fix travis build error

* VPN profile bugfix

* Fix casing for NRP automapping

* Fix parsing errors

* remove BgpSettings from P2SVpnGateway object

* Added AddressPrefixes on Subnet properties (#3714)

* Added address prefixes to subnet

* Added example of multiple address prefixes.

* Python Network conf 2018-08-01

* Changes to include required properties for outbound rule (#3728)

* Mark outbound rule properties as required

* Update outbound rule with required properties

* Removing old change

* Fix Python 2018-08 conf

* Fix swagger contract errors with NRP

* Fix network swagger

* Fix virtualHub Properties for SDK and PS JSON serializer compliance

* Fix bugs

* Changing Cortex swagger to match NRP

* Bug fixes

* bugFix3

* Bug fix

* Use only PublicIpAddress in Azure Firewall, not both Public and InternalPublic (#3743)

* Do not confuse users with both InternalPublicIp and PublicIp for the same thing

* Update AzureFirewall PUT example

* This closes #3474

* Use the correct name for publicIPAddress in AzureFirewall PUT example

* Azure Firewall FQDN Tag top level resource (#3744)

* Azure Firewall FQDN Tag top level resource

* Name FqdnTags property in AzureFirewallApplicationRule properly

* Empty commit to trigger new validation

* Vtapfinal (#3684)

* Swagger Coverage/Completeness- for operation CheckNameAvailability (#3663)

* Update service.json

* CheckNameAvailability

* FIxed Parameter name

fixed couple of parameters name

* Adds base for updating Microsoft.Network from version stable/2018-07-01 to version 2018-08-01

* Updates readme

* Updates API version in new specs and examples

* vtap update

* VTAP changes

* VTAP changes

* Adding examples for VTAP resource

* Update changes for Examples

* Further fixes to examples

* Add exception for vtap's RequiredPropertiesMissingInResourceModel in readme

* Azure Firewall NAT Rule Collection (#3745)

* NAT Rule Collections for Azure Firewall

* Update examples to include the new NAT Rule Collection

* Shorten the NAT RC Properties name and mark etag read-only

* Add NAT Rule Collection to Azure Firewall properties

* Fix all <<a Azure>> in helper messages

* Do not allow ICMP as Network protocol (#3742)

* Adding linkedResourceType and hostedWorkloads to nic (#3780)

* Adding support for interface endpoints (#3783)

* Creating Interface Endpoint definition

* Adding references to interface endpoints in existing resources

* Added Trusted root certificate changes to new API version (#3731)

* Add ExpressRoute gateway resource (#3776)

* Initial version

* Refactor connections.

* Update API version

* Add enum values.

* Addressed all comments.

* Adding support for delegations on a subnet (#3805)

* Adding subnet delegation property on VNET

* Updating subnet/vnet create & get examples to reflect delegation

* Adding support for GetAvailableDelegations operation

* Fixing properties in Swagger & updating examples to reflect fixes

* Add availableDelegation as a subResource

* Remove type from examples

* Further updates to delegation PR

* Additional changes to structure

* Correct issues for SDK generation (#3818)

* Remove x-ms-client-flatten attribute as that results in wrong
descriptiosn during generation of the SDKs.

* Missing descriptions.

* Rename TargetUrls to TargetFqdns (#3739)

* Fix travis errors in examples

* Fixed error examples for P2S cortex resources

* Fix swagger

* Fix Semantic validation error

* network swagger fix

* Fix travis

* Fix travis errors

* Revert version change

* Dummy changes for CI restart

* Swagger changes (#3857)

* Relocated readOnly property

* Readded provisioning state to interfaceEndpoint.json

* Updated networkInterface.json

* Updated serviceEndpointPolicy.json

* Update virtualNetworkTap.json

* Fix semantic issue

* Multiple changes

* More example fixes

* Dummy commit

* Fixed duplicate

* changes to swagger in interface endpoints and subnet delegation (#3860)

* Revert "Trusted Root certificate (#3668)"

This reverts commit 3ff4b520bb59cccb04cde1a0b2dcebabfd29a4b0.

* fixed api version in azure firewall example

* Review comments incorporation (#3859)

johnemmanuelm committed:
StorSimple1200Series Azure Rest API Spec (#3718)

* StorSimple1200Series: Swagger spec and json

* Second version

* Fixing json eof file error

* Fixed ref path example json

* Added deactivate, delete examples for device and managers.

* Missing eof json error

* Fixing syntax error

* Revert last 3 changes to swagger

* fixing Json format error

* Added more examples

* Added DevicesFailover example

* Fixing more examples

* Added description to definitions

* Fixed duplicate param errors in swagger submitted in the prev commit

* Added default response to all paths.
CR  comment ref: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#default-response

* Adding readme.md

* Adding readme.md for nodejs and ruby

* Fixing folder names as case senstive

* Reverting all readme.md changes except input file

* Fixing linter errors

* Fixed linter errors

* Addressed more Linter errors

* Fixed linter errors

* Added more examples and fixed linter errors

* Fixing typo in an example json

viveklingaiah committed:
[GuestConfiguration] Move delete assignment path for now, correct sdk destination folder (#3865)

kalyanaj committed:
EventGrid: Updated README.MD configuration to include the new preview API version. (#3830)

* 1) Fix for linter error "Properties of a PATCH request body must not be default-valued. PATCH operation: 'EventSubscriptions_Update' Model Definition: 'EventSubscriptionUpdateParameters' Property: 'eventDeliverySchema'".
2) Updated the default value of EventDeliverySchema to the correct value used by the service.

* Initial version of 2018-09-15-preview API version (clone of 2018-05-01-preview to make it easier to view the diffs).

* Swagger changes for 2018-09-15-preview API version.

* Update README.MD configuration for EventGrid to include the new preview API version.

* Added two new operatorTypes to AdvancedFilter + marked a couple of properties readOnly.

Commit:
Remove package-version from readme.nodejs.md file (#3872)

* Remove package-version from readme.nodejs.md file

* Update azure-arm-storsimple1200series package-name and output-folder

alexeldeib committed:
[App Insights/Operational Insights] Add GET /query operation, fix operationId inconsistency (#3771)

* Add query get for completeness; fix operationId inconsistency

* fix params

* remove accidental param

* fix timespan param in GET

KrisBash commented:
@quintinb 

Can you schedule a reviewing meeting with the API review board? That would help expedite from the ARM side. Thanks

jhendrixMSFT committed:
Fix Go SDK generation (#3869)

Added package-2018-09 to Go SDK.
Moved Go SDK config into its own file.

grimurjonsson committed:
NetApp adding subnetId to volume properties (#3873)

sergey-shandar committed:
Update ADHybridHealthService.json (#3874)

quintinb commented on specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AccountFilters.json:
```diff
@@ -0,0 +1,553 @@
+      "type": "object",
+      "required": [
+        "property",
+        "operation"
```
Seems reasonable.  I made the value required as well.

quintinb commented on specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AccountFilters.json:
```diff
@@ -0,0 +1,553 @@
+          },
+          "description": "The track selections."
+        }
+      },
```
No, you are correct there isn't a scenario where you would provide an empty track selections collection.  I added the required attribute.

quintinb commented on specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/AccountFilters.json:
```diff
@@ -0,0 +1,553 @@
+        }
+      }
+    }
+  },
```
Correct.  AccountFilters are just DB write operations so there is no need for an LRO.

Commit:
ServiceBus: 2018-01-01 Preview - for managing Namespace, IPFilter Rules, VirtualNetworkRules and Zone Redundant (#3754)

* 2018 Preview

* changed the preview swagger name

* corrected swagger

* fixed validation errors

* corrected the type in VirtualNetworkRuleListResult

* updated swagger, removed the opreations API

* preview move to the begining of the file

* fixed casing

elemount committed:
Add mariadb swagger spec (#3797)

* Add mariadb swagger spec

* Fix readme and set properties

Commit:
Swagger spec for disk.json + compute.json 2018-06 (#3854)

* removing resourceUpdate, adding contents to disk and snapshot

* adding swagger spec for disk.json 2018-06-01

* Disks 2018-06-01 for Python

* update disk

* Update readme.md

* adding compute.json changes

* deleting duplicates

* Fix indent

* adding suppress

* Update suppression rules

lmazuel committed:
Update .travis.yml (#3888)

Commit:
Generate readme.md file for azure-arm-mariadb until one is provided (#3893)

jhendrixMSFT committed:
Fix Go SDK codegen for mariadb (#3892)

Moved preview package under preview subdirectory.
Moved Go SDK config to its own config file.

t-robalt committed:
[DO NOT MERGE] Added list support to VA policies (#3878)

* Added list support to VA policies

* Changed suffixes

* added missing examples

* Updated database example

* Changed operation name.

the suffix of "_ListBy..." should be a substring of the operation group, thus changed to Instance.

asinn826 committed:
Fix incorrect url and clarify subscription-level disable (unsure why this came with so many commits) (#3886)

* add serial console reference

* update version

* Add readme.md

* update readme.md

* add config to readme

* update default parameter

* fix json

* fix camelcase error

* add x-ms-examples

* add renamed DisableConsoleExample.json

* fix json

* add operations, address github comments

* update operations path spec

* add more examples, some minor changes

* update x-ms-parameter-location

* trivial modification to error message to kickstart a new build process

* example updates

* update example again

* update api version

* modify api version reference

* one more api version reference

* (hopefully) final api version reference update

* update examples and codeowners

* trivial change to kickstart build again

* add python

* trivial change to kickstart build again

* Update CODEOWNERS

* clarify the subscription-level disable

* fix incorrect url

hosunmsft committed:
Add degreeOfParallelismPercent to ADLA Job Contract (#3871)

* Add degreeOfParallelismPercent to JobInformationBasic

* Add degreeOfParallelismPercent to JobInformationBasic

Commit:
Update the travis.yml file to do a TypeScript generator check (#3902)

vikram-m committed:
AlertsManagement RP GA version (#3894)

* Adding GA stable 2018-05-05 version

Moving from public preview to GA version

* Modifying the pointers to stable version

* Fixing the linter diff errors

* Fixing linter error - duplicate parameter

* Fixing model validator errors

* Syntax error in response

Commit:
Fix to version in examples and in runbook.json (#3899)

bganapa committed:
Fixing the name of the profile, matching to the file name (#3897)

vickyv51 committed:
Publish public preview swagger spec for "Azure metrics Ingestion REST  API" (#3856)

* -Publish "Azure metrics REST Ingestion API" public preview swagger spec

* Address review comments
 - Use a shorter operationId
 - Remove any non-success response code with default response clause
 - Add a new example and reference it in the swagger spec
 - Include readme.md file for the swagger spec

* Fix a bug in readme.md.
 - Inline the readme.python.md file in the readme.md

* -Remove validation section that is not necessary

* -Address review comments & fix several copy paste mistakes to use sensible names instead

* Address the PR comments
  - Fix the readme.md naming incompatibilities
  - Remove the c#-ish namespace prefix

* -Remove assembly namespace prefix in few other places
-Rename bunch of properties to follow camel case naming
-Update the enum types to have the "x-ms-enum" part of the json

* -Fix last batch of ARM Errors
-Update path and namespace for Java section

* -Drop all the types that recursively came with ObjectResult and use simple object type instead. This should make the swagger spec clean and more easily maintainable in the long run

* -Fix the example to just follow key-value pairs for all required parameters.

* -Fix the responses section in the example

* -Fix the type bug in the example, use int32 instead of string

Flip-Liquid committed:
Add 202 response code as expected for network profile deletion (#3907)

* add 202 response code as expected for network profile deletion

add 202 to delete example

* added 200 response code for network profile

blueww committed:
[Storage] add new stroage account kind "FileStorage" and "BlockBlobStorage" and new feature fileaadIntegration (#3864)

* [Storage] Add api version 2018-07-01

* [Storage] update api version 2018-07-01

* [Storage] Add new Storage Account Kind and Sku

* [Storage] Remove the sku Premium_ZRS change as it won't be release in Ignite

* [Storage] Support FileAadIntegration

* Storage Python conf

* [Storage] modify per comments

* [Storage] Add the parameter location changes in 2018-03-01-preview to 2018-07-01

* [Storage] Add back Management Policy from 2017-03-01-preview

* [Storage] remove global usage as server change

* [Storage] Add new Sku Premium_ZRS

* [Storage] move Management policy swagger to 2018-03-01-preview folder

quintinb committed:
Updates from SDK team review feedback

quintinb committed:
Updates from SDK team review feedback

quintinb commented on specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Accounts.json:
```diff
@@ -0,0 +1,910 @@
+              },
+              {
+                "value": "Total"
+              }
```
Added.  The team wanted to show a difference between the count of items (say the number of requests) versus the total of a parameter (total bytes delivered).  These are both summations so it isn't my favorite design but I added some comments to try to make the distinction clearer.

quintinb committed:
Adding missing required annotation in FilterTrackPropertyCondition

quintinb committed:
Adding missing required annotation in FilterTrackPropertyCondition

quintinb committed:
Added required annotation to the FilterTrackSelection

quintinb committed:
Added required annotation to the FilterTrackSelection

refortie committed:
Remove default response from Sessions to bring them inline with the remaining artifacts (#3868)

Commit:
New tag for 2018-06 APIs for Automation (#3906)

* Fix to version in examples and in runbook.json

* Added new 2018-06-30 tag to readme and other nit fixes

* Added missing property to make JobStreamProperties definitions identical

Commit:
Add readme.typescript.md for DNS (#3903)

* Add readme.typescript.md for DNS

* Add azure-sdk-for-js swagger-to-sdk entry in DNS readme.md

vickyv51 committed:
Fix the Azure metrics data-plane api swagger spec to ensure the documentaiton generation works as expected (#3911)

* -Publish "Azure metrics REST Ingestion API" public preview swagger spec

* Address review comments
 - Use a shorter operationId
 - Remove any non-success response code with default response clause
 - Add a new example and reference it in the swagger spec
 - Include readme.md file for the swagger spec

* Fix a bug in readme.md.
 - Inline the readme.python.md file in the readme.md

* -Remove validation section that is not necessary

* -Address review comments & fix several copy paste mistakes to use sensible names instead

* Address the PR comments
  - Fix the readme.md naming incompatibilities
  - Remove the c#-ish namespace prefix

* -Remove assembly namespace prefix in few other places
-Rename bunch of properties to follow camel case naming
-Update the enum types to have the "x-ms-enum" part of the json

* -Fix last batch of ARM Errors
-Update path and namespace for Java section

* -Drop all the types that recursively came with ObjectResult and use simple object type instead. This should make the swagger spec clean and more easily maintainable in the long run

* -Fix the example to just follow key-value pairs for all required parameters.

* -Fix the responses section in the example

* -Fix the type bug in the example, use int32 instead of string

* -Fix the swagger spec to ensure the documentation content is rendered as expected.

* -Fix the readme.md file to refer the correct file

* -Delete the old spec as it is now renamed

* -Call operationId with suffix Create to be consistent
-Rename the file to be consistent with operationId

noferrar committed:
Add specifications for dynanic threshold GA api (#3883)

* Add specifications for dynanic threshold GA api

* Change to capital letter and capitalizing abbreviations

* Add metricnamespace & $filter parameters

tomerlmsft committed:
Introducing new API for Advanced Threat Protection (#3306)

* Introducing new API for Advanced Threat Protection

API form -
{resourceUri}/providers/Microsoft.Security/advancedThreatProtectionSettings/{settingName}

Suppporting verbs -
GET, PUT and Delete.

URI example -
/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/resourceGroups/SampleRG/providers/Microsoft.Storage/storageAccounts/samplestorageaccount/providers/Microsoft.Security/advancedThreatProtectionSettings/current

Design doc for ATP -
https://microsoft.sharepoint.com/teams/Data_Security/Documents/Dev%20Specs/Storage%20Threat%20Detection/Threat%20Detection%20for%20Azure%20Storage%20-%20Design%20Document.docx?d=w996c55b5a160468c9a9bd0434418a4d0

* Addressing pull request comments

*Removed DELETE operation, stayed only with GET and PUT.
*Changed ResourceUri to ResourceId
*Changed isEnabled to enabled

* Renaming obejct from AdvancedThreatProtectionSettings to AdvancedThreatProtectionSetting

* Removed delete operation from the swagger definitions

* Fixed test failures

* Changing "AdvancedThreatProtectionSettingName" parameter to enum type

* Changed param name from 'enabled' to 'isEnabled'

t-arrinb committed:
Add algorithm types to threat detection disabled alerts description (#3876)

Added Data_Exfiltration and Unsafe_Action as allowed values to disabled alert.

MattSable committed:
Add Batch AI workspace property to machineLearningServices swagger (#3910)

* Add BatchAIWorkspace to main swagger and examples

* Fix whitespace

* Update casing to match swagger definition

* Correctly update casing of all usages of batchaiWorkspace

KrisBash commented:
Discussed changes with @quintinb . No further review meeting required from ARM

KrisBash reviewed (approved):
LGTM

LanaShafik committed:
Adding app version settings APIs to swagger file. (#3850)

Commit:
Add generate-metadata: true to dns readme.typescript.md (#3920)

michimune committed:
Add networkTrace resource and APIs (#3811)

* Add networkTrace resource and APIs

* Add examples of network trace APIs

* Add x-ms-examples to network trace APIs

* Fix file name typo of GetWebSiteNetworkTraces.json

* Fix syntax error in examples/GetWebSiteNetworkTraces.json

* Fix more validation errors

* Fix StopWebSiteNetworkTrace.json

* Fix network trace APIs

ankushbindlish2 committed:
Add creation parameters to StorageSync ResourceProvider for SDK (#3863)

* Add creation parameters

* Example update

* Fix RegisteredServer

Commit:
Add readme.typescript.md for compute (#3924)

* Add readme.typescript.md for compute

* Change readme.typescript.md header to TypeScript

Commit:
Create arm-network readme.typescript.md file (#3932)

* Create arm-network readme.typescript.md file

* Add generate-metadata: true to network/resource-manager readme.typescript.md file

daozha committed:
Update Microsoft.Backup.Admin API version 2018-09-01 (#3891)

* Microsoft.Backup.Admin copy API version 2018-09-01 from 2016-05-01

* Update Microsoft.Backup.Admin API version 2018-09-01

* resolve comments

hosunmsft committed:
Add degree of parallelism percent to ADLA Job PREVIEW API (#3904)

* Add degreeOfParallelismPercent to JobInformationBasic

* Add degreeOfParallelismPercent to JobInformationBasic

* add degreeOfParallelism to preview API
:wq
:wq


ankushbindlish2 committed:
StorageSync : Clone existing ApiVersion 2018-04-02 for creating a new version 2018-07-01 (#3927)

* Clone existing ApiVersion 2018-07-01

* Change folder structures and remove versions from readme*

* Review comments incorporation.

* Add go

* one tag only

* fix version

quintinb committed:
Fixing merge conflict

daschult commented on specification/datalake-analytics/data-plane/readme.nodejs.md:
````diff
@@ -24,4 +25,7 @@ nodejs:
   source-code-folder-path: lib/job
-```
\ No newline at end of file
+  generate-license-txt: true
````
Revert the changes in this file. I just made these changes and this PR is attempting to undo them.

quintinb replied:
I think I messed up a merge.  Please stand by...

daschult commented on .travis.yml:
```diff
@@ -47,7 +53,7 @@ script:
   - >-
     if [[ $MODE == 'java' ]]; then
-      travis_wait 30 scripts/swagger-to-sdk.sh Azure/azure-libraries-for-java -v
+      travis_wait 30 scripts/swagger-to-sdk.sh Azure/azure-sdk-for-java -v
```
I know that azure-libraries-for-java is correct, so this is probably wrong.

anuchandy commented:
Closing this PR in-favor of https://github.com/Azure/azure-rest-api-specs/pull/3980

AutorestCI commented:
# Automation for azure-sdk-for-js
